org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/configuration/ConfigurationManager.java

```diff
@@ -49,7 +50,11 @@ BuildConfiguration createBuildConfiguration(File rootProjectDirectory, boolean o
 
     void deleteProjectConfiguration(IProject project);
 
-    RunConfiguration loadRunConfiguration(ILaunchConfiguration configuration);
+    CompositeConfiguration loadCompositeConfiguration(IWorkingSet workingSet);
```

Comment:
❌ We should not use `IWorkingSet` here as it comes from a UI plugin. The method should take a generic supplyer argument.